### PR TITLE
Event additions

### PR DESCRIPTION
**Pull request by supercrafter100**
Added 3 more events:

- userGroupAdded
- userGroupRemoved
- topicReply

### CODE REVIEW

File: core/classes/Core/User.php [hunk 1]
```diff
@@ -718,6 +725,13 @@ public function removeGroup(?int $group_id): bool {
             ]
         );
 
+        EventHandler::executeEvent("userGroupRemoved", [
```
Comment (Aberdeener) [inline core/classes/Core/User.php:728]: Single quotes please

File: core/classes/Core/User.php [hunk 2]
```diff
@@ -181,6 +181,13 @@ public function addGroup(int $group_id, int $expire = 0, $group_data = null): bo
             $this->_groups[$group_id] = new Group($group_data);
         }
 
+        EventHandler::executeEvent("userGroupAdded", [
```
Comment (Aberdeener) [inline core/classes/Core/User.php:184]: Single quotes please

**Review by Aberdeener — CHANGES_REQUESTED** [review]

File: modules/Forum/module.php [hunk 3]
```diff
@@ -59,7 +59,7 @@ public function __construct(Language $language, Language $forum_language, Pages
         EventHandler::registerEvent('newTopic',
             $this->_forum_language->get('forum', 'new_topic_hook_info'),
             [
-                'uuid' => $this->_language->get('admin', 'uuid'),
+                'user_id' => $this->_language->get('admin', 'user_id'),
```
Comment (Aberdeener) [inline modules/Forum/module.php:62]: Why was this changed?

**Review by Aberdeener — APPROVED** [review]

File: modules/Forum/module.php (hunk 3 above)
Comment (supercrafter100) [reply modules/Forum/module.php:62]: Uuid wasn't provided when the event was triggered but the user id.

Comment (supercrafter100) [reply modules/Forum/module.php:62]: As you can see on line 205 here https://github.com/NamelessMC/Nameless/blob/v2/modules/Forum/pages/forum/new_topic.php

File: modules/Forum/pages/forum/view_topic.php [hunk 4]
```diff
@@ -293,6 +293,24 @@
                 'topic_reply_date' => date('U')
             ]);
 
+            // Fire event
+            EventHandler::executeEvent('topicReply', [
```
Comment (partydragen) [inline modules/Forum/pages/forum/view_topic.php:297]: Can you include topic id and post id

Comment (partydragen) [reply modules/Forum/pages/forum/view_topic.php:297]: Post id is not forum id xD
